core/src/main/scripts/importer/validateData.py

```diff
@@ -1312,12 +1331,24 @@ def checkHeader(self, cols):
         num_errors = super(ClinicalValidator, self).checkHeader(cols)
 
         if self.numCols != len(self.attr_defs):
-            self.logger.error(
-                'Varying numbers of columns in clinical header (%d, %d)',
-                len(self.attr_defs),
-                len(self.cols),
-                extra={'line_number': self.line_number})
-            num_errors += 1
+             if not self.relaxed_mode:
+                self.logger.error(
+                    'Varying numbers of columns in clinical header (%d, %d)',
+                    len(self.attr_defs),
+                    len(self.cols),
+                    extra={'line_number': self.line_number})
+                num_errors += 1
+            
+        # fill in missing attr_defs data if in relaxed mode and clinical 
+        if self.fill_in_attr_defs:
+            self.logger.info('Filling in missing attribute properties for clinical data.')
+            missing_attr_defs = {}
+            for col_index, col_name in enumerate(cols):                    
+                missing_attr_defs[col_index] = {'display_name': col_name,
+                                         'description': col_name,
+                                         'datatype': 'STRING',
+                                         'priority': '1'}
```

Comment:
The reason for this is that if you do set the datatype that will actually be used when loading, the values in the file can be validated accordingly and numeric attributes really be numeric (or null) values. Certain range sliders in the frontend, for instance, will break otherwise.